### PR TITLE
add form option

### PR DESCRIPTION
**Pull request by Eomm**
I would like to introduce this option to easy the `form` usage when `inject` with fastify

### CODE REVIEW

**Comment by Eomm** [comment]
Could the solution be to add the form dependency optional in the package json?

I'm targeting to have a fast way to send a form as got does

**Comment by mcollina** [comment]
setting it as optional would still have npm download it :/.

**Comment by Eomm** [comment]
> setting it as optional would still have npm download it :/.

optional by input like it was done in fastify-compress for the brotli support could be fine for you?
So it must be a user action activate this syntactic sugar

example:

```
  const form = await inject(dispatch, {
    method: 'post',
    url: '/upload',
    form: require('form-data'),
    payload: {
      myField: 'hello',
      myfile: fs.createReadStream(`./path/to/file`)
    }
  })
```

**Comment by mcollina** [comment]
that works, yes. However I think this would be slightly better and it does not require any change here, minus a quick addition to the README:

```js
const form = require('simple-form')
const res = await inject(dispatch, {
    method: 'post',
    url: '/upload',
    payload: form({
      myField: 'hello',
      myfile: fs.createReadStream(`./path/to/file`)
    })
  })
```

**Comment by Eomm** [comment]
I didn't find `simple-form` or `form-data` support this syntax:

```
    payload: form({
      myField: 'hello',
      myfile: fs.createReadStream(`./path/to/file`)
    })
```

Are you suggesting a little new module?

**Comment by mcollina** [comment]
Yes i am ;).

**Comment by Eomm** [comment]
> Yes i am ;).

Before I publish it I would really appreciate if you could give it a check

https://github.com/Eomm/form-method#form-method

**Comment by mcollina** [comment]
I would just return an object like `{ payload, headers }`. I would also stick to a stream for the payload, it seems easier for the recipient. You can get one with `Readable.from(string)`.

Also, I'm unsure about the name, why `method`?

**Comment by Eomm** [comment]
>  I would also stick to a stream for the payload, it seems easier for the recipient

Thanks, I agree, uniformed the output 👍 

> why method?

I was thinking to `<form method=POST>`... and it sounds better than `form-enctype` 😅
I found these two names free on npm
Is it really so bad?

**Comment by mcollina** [comment]
It’s unclear :(. I would go with something with simple or auto in the name.

**Comment by Eomm** [comment]
Renamed `form-auto-content` 👍

PR updated

File: README.md [hunk 1]
```diff
@@ -72,21 +72,23 @@ try {
...
+    myFile: fs.createReadStream(`./path/to/file`)
+  })
 
   const res = await inject(dispatch, {
     method: 'post',
     url: '/upload',
-    payload: form,
-    headers: form.getHeaders()
+    payload: form.payload,
+    headers: form.headers
```
Comment (mcollina) [inline README.md:91]: I would use `...form` instead, is more idiomatic

**Pull request merged**